superset/models/dashboard.py

```diff
@@ -423,11 +424,6 @@ def export_dashboards(  # pylint: disable=too-many-locals
                 remote_id=eager_datasource.id,
                 database_name=eager_datasource.database.name,
             )
-            datasource_class = copied_datasource.__class__
```

Comment:
This was breaking the serialization of the datasources, i.e., JSON detected a circular reference. Given that the datasource children (metrics, columns) are defined via the `sqlalchemy.orm.relationship` this logic no longer seems needed.